marlowe-dashboard-client/src/Dashboard/State.purs

```diff
@@ -95,80 +92,29 @@ handleAction ::
-  handleAction input $ WalletIdInputAction $ InputField.SetValidator $ walletIdError remoteWalletInfo walletLibrary
-
-handleAction input (SaveNewWallet mTokenName) = do
-  oldWalletLibrary <- use _walletLibrary
-  walletNickname <- use (_walletNicknameInput <<< _value)
-  walletIdString <- use (_walletIdInput <<< _value)
-  remoteWalletInfo <- use _remoteWalletInfo
-  let
-    mWalletId = parsePlutusAppId walletIdString
-  case remoteWalletInfo, mWalletId of
-    Success walletInfo, Just walletId -> do
-      handleAction input CloseCard
-      let
-        -- note the empty properties are fine for saved wallets - these will be fetched if/when
-        -- this wallet is picked up
-        walletDetails =
-          { walletNickname
-          , companionAppId: walletId
-          , marloweAppId: PlutusAppId emptyUUID
-          , walletInfo
-          , assets: mempty
-          , previousCompanionAppState: Nothing
-          }
-      modifying _walletLibrary (insert walletNickname walletDetails)
-      insertIntoWalletLibrary walletDetails
-      newWalletLibrary <- use _walletLibrary
-      -- if a tokenName was also passed, we need to update the contract setup data
-      for_ mTokenName \tokenName -> do
-        handleAction input $ TemplateAction Template.UpdateRoleWalletValidators
-        handleAction input $ TemplateAction $ Template.RoleWalletInputAction tokenName $ InputField.SetValue walletNickname
-    -- TODO: show error feedback to the user (just to be safe - but this should never happen, because
-    -- the button to save a new wallet should be disabled in this case)
-    _, _ -> pure unit
+handleAction _ (WalletDataAction walletDataAction) = case walletDataAction of
+  WalletData.CancelNewContactForRole -> assign _card $ Just ContractTemplateCard
+  _ -> toWalletData $ WalletData.handleAction walletDataAction
 
 handleAction _ ToggleMenu = modifying _menuOpen not
 
 handleAction input (OpenCard card) = do
-  case card of
-    SaveWalletCard _ -> do
-      walletLibrary <- use _walletLibrary
-      assign _remoteWalletInfo NotAsked
-      handleAction input $ WalletNicknameInputAction InputField.Reset
-      handleAction input $ WalletNicknameInputAction $ InputField.SetValidator $ walletNicknameError walletLibrary
-      handleAction input $ WalletIdInputAction InputField.Reset
-      handleAction input $ WalletIdInputAction $ InputField.SetValidator $ walletIdError NotAsked walletLibrary
-    _ -> pure unit
-  assign _card $ Just card -- explain
+  -- first we set the card and reset the contact and template card states to their first section
+  -- (we could check the card and only reset if relevant, but it doesn't seem worth the bother)
+  modify_
+    $ set _card (Just card)
+    <<< set (_walletDataState <<< _cardSection) Home
+    <<< set (_templateState <<< _contractSetupStage) Start
+  -- then we set the card to open (and close the mobile menu) in a separate `modify_`, so that the
+  -- CSS transition animation works
```

Comment:
Not sure if it makes a difference, but notice that now you have `Halogen.Animation (waitForAllAnimations)` to make sure that all animations and transitions on an element have finished

Reply:
That might be better. The tricky thing will be making the card animation responsive - it needs to come up from the bottom on small screens, and over from the right on large screens. That's easy if it's just done with CSS transitions, but I'm not immediately sure how to do it with animations... :thinking: